### PR TITLE
Remove old razor

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Remove usages of old razor.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Microsoft.AspNetCore.Mvc.TagHelpers.csproj [hunk 1]
```diff
@@ -16,6 +16,7 @@
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.Razor\Microsoft.AspNetCore.Mvc.Razor.csproj" />
 
+    <PackageReference Include="Microsoft.AspNetCore.Razor" Version="$(AspNetCoreVersion)" />
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Microsoft.AspNetCore.Mvc.TagHelpers.csproj]: Why is this needed?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Microsoft.AspNetCore.Mvc.TagHelpers.csproj]: Same question for others below

**Comment by ryanbrandenburg** [comment]
Looks like some tests such as `RazorEngineTest.RazorEngine_Model_DesignTime` are failing [here](https://travis-ci.org/aspnet/Mvc/jobs/211938944). Do I just have to change these numbers to meet the new ones being output?

**Comment by ajaybhargavb** [comment]
Rebasing with dev should fix it 👍

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Microsoft.AspNetCore.Mvc.TagHelpers.csproj (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Microsoft.AspNetCore.Mvc.TagHelpers.csproj]: These assemblies are still needed for Tag Helpers. What's important is that we need to remove usage of the old compiler types.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs [hunk 2]
```diff
@@ -10,22 +10,18 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
     public class CompositeTagHelperDescriptorResolver : ITagHelperDescriptorResolver
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs]: Would we still need this type if all its doing is delegating to the inner one?

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/DependencyInjection/MvcRazorMvcBuilderExtensionsTest.cs [hunk 3]
```diff
@@ -36,8 +36,9 @@ public void AddTagHelpersAsServices_ReplacesTagHelperActivatorAndTagHelperTypeRe
             var activatorDescriptor = Assert.Single(services.ToList(), d => d.ServiceType == typeof(ITagHelperActivator));
             Assert.Equal(typeof(ServiceBasedTagHelperActivator), activatorDescriptor.ImplementationType);
 
-            var resolverDescriptor = Assert.Single(services.ToList(), d => d.ServiceType == typeof(ITagHelperTypeResolver));
-            Assert.Equal(typeof(FeatureTagHelperTypeResolver), resolverDescriptor.ImplementationType);
+            // This was removed to facilitate the move to new razor and will be re-added as part of https://github.com/aspnet/Mvc/issues/5768
+            //var resolverDescriptor = Assert.Single(services.ToList(), d => d.ServiceType == typeof(ITagHelperTypeResolver));
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/DependencyInjection/MvcRazorMvcBuilderExtensionsTest.cs]: Remove it for now. It's in the source control history if you need to look it up. Let's not leave commented out code.

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/Microsoft.AspNetCore.Mvc.TagHelpers.Test.csproj [hunk 4]
```diff
@@ -12,6 +12,7 @@
     <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.TestCommon\Microsoft.AspNetCore.Mvc.TestCommon.csproj" />
 
     <PackageReference Include="Microsoft.AspNetCore.Http" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Razor" Version="$(AspNetCoreVersion)" />
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/Microsoft.AspNetCore.Mvc.TagHelpers.Test.csproj]: Why does the test need an explicit reference?

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs (hunk 2 above)
Comment (ryanbrandenburg) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs]: I don't think so, but it's my understanding that we'll be adding the ViewComponentTagHelperDescriptorResolver back in, so I figured we didn't want to do too much work rearchitecting this in the meantime.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs]: The `ITagHelperDescriptorResolver` type here is going to be removed. Let's delete this and bring it back in Razour.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompositeTagHelperDescriptorResolver.cs]: We will need to remove this because `ITagHelperDescriptorResolver` is being removed
